run_robot_test.sh

```diff
@@ -71,29 +75,135 @@ while [ "$#" -gt 0 ]; do
+    OPTSTR="d:e:f:hi:r:st:v:"
+    while getopts $OPTSTR flag; do
+        case "$flag" in
+            d) # Specify directory to store artifacts and reports from each test run
+                TEST_ARTIFACT_DIR=$OPTARG
+                create_artifact_dir
+            ;;
+            e) # Specify excluded tags
+                TEST_EXCLUDE_TAG=$OPTARG
+            ;;
+            f) # Specify the test variable file
+                TEST_VARIABLES_FILE=$OPTARG
+            ;;
+            h) # Display Usage and exit
+                disp_usage $OPTSTR
+                exit 0
+            ;;
+            i) # Specify included tags
+               # Example: sanityANDinstall sanityORinstall installNOTsanity
+                TEST_INCLUDE_TAG=$OPTARG
+            ;;
+            r) # Additional arguments to pass to the robot cli
+                EXTRA_ROBOT_ARGS=$OPTARG
+            ;;
+            s) # Skip the pip install during the execution of this script
+                SKIP_PIP_INSTALL=1
+            ;;
+            t) # Specify test case to run
+                TEST_CASE_FILE=$TEST_CASE_LOCATION/$OPTARG
+            ;;
+            v) # Override/Add global variables specified in the test variables file
+                TEST_VARIABLES="${TEST_VARIABLES} --variable $OPTARG"
+            ;;
+            *)
+                echo "Invalid Input"
+                disp_usage $OPTSTR
+                exit 1
+            ;;
+        esac
+    done
+
+    shift $((OPTIND -1))
+
+    # ------ Validate Inputs --------
+    if [[ ! -f "${TEST_VARIABLES_FILE}" ]]; then
+        echo "---- Robot Framework test variable file (test-variables.yml) is missing"
+        exit 1
+    fi
+}
+
+
+disp_usage() {
+    echo
+    echo "Usage:    $0 [$1]"
+    echo "Possible inputs: -d -e -f [-h] -i -r [-s] -t -v"
+    echo "Specified Inputs: $SPECIFIED_INPUTS"
+    echo "Explanation of inputs "
+    echo "      -d <TEST_ARTIFACT_DIR>: Specify directory to store artifacts and reports from each test run"
+    echo "      -e <TEST_EXCLUDE_TAG>: Specify excluded tags for Robot"
+    echo "      -f <TEST_VARIABLES_FILE>: Specify the test variable file"
+    echo "      -h: Display Usage and exit"
+    echo "      -i <TEST_INCLUDE_TAG>: Specify included tags for Robot"
+    echo "      -r <EXTRA_ROBOT_ARGS>: Additional arguments to pass to the robot cli"
+    echo "      -s: Skip the installation of required Python Libraries during the execution of this script"
+    echo "      -t <TEST_CASE_FILE>: Specify test case to run, should be present under Tests directory"
+    echo "      -v <>: Override/Add global variables specified in the test variables file"
+}
+
+
+pip_install() {
+    # This is optional so we are not creating/updating the virtualenv everytime we run a test
+    echo "--------- Installing required Python Libraries ----------"
+    VENV_ROOT=${PWD}/venv
+    # setup virtualenv
+    python3 -m venv ${VENV_ROOT}
+    source ${VENV_ROOT}/bin/activate
```

Comment:
IMO, we should avoid doing this step as below we are referencing the binaries on that environment directly.

Reply:
So the virtualenv setup should be made an optional one and should not be updated with every run
From the commented line at 155, it seems like this was a TO-DO item for ODS team

I have modified this part to setup the virtualenv only when necessary.
It will be more clear with the updated PR